tests/callbacks/TestWorkload.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef KOKKOS_UTILS_TESTS_CALLBACKS_TESTWORKLOAD_HPP
+#define KOKKOS_UTILS_TESTS_CALLBACKS_TESTWORKLOAD_HPP
+
+#include "Kokkos_Core.hpp"
+#include "Kokkos_Profiling_ProfileSection.hpp"
+
+#include "kokkos-utils/concepts/ExecutionSpace.hpp"
+
+namespace Kokkos::utils::tests::callbacks
+{
+
+template <Kokkos::utils::concepts::ExecutionSpace Exec>
+struct MyWorkload
+{
+    struct MyFunctor
+    {
+        template <std::integral T>
+        KOKKOS_FUNCTION
+        void operator()(const T) const {}
+    };
+
+    void execute(const Exec& exec) const
+    {
+        Kokkos::Profiling::ProfilingSection profile_section("profile section");
+
+        profile_section.start();
+
+        Kokkos::Profiling::pushRegion("computation - level 0");
+
+        Kokkos::parallel_for(
+            "computation - level 0 - pfor",
+            Kokkos::RangePolicy(exec, 0, 1),
+            MyFunctor{}
+        );
+        exec.fence("computation - level 0 - fence after pfor");
+
+        Kokkos::Profiling::pushRegion("computation - level 1");
+
+        // Another kernel, a parallel reduce on the default execution space instance.
+        double my_result;
+        Kokkos::parallel_reduce(
+            "computation - level 1 - preduce on default exec",
+            Kokkos::RangePolicy(0, 1),
+            KOKKOS_LAMBDA (const typename Exec::size_type, const double&) {},
+            my_result
+        );
+
+        Kokkos::Profiling::popRegion(); // computation - level 1
```

Comment:
Use `ScopedRegion` instead ?

Reply:
OK. Indeed. I've kept the "level 0" as push/pop. But then I've changed the "level 1" into "Scoped".